### PR TITLE
[16.0] [ADD] purchase_sign

### PR DESCRIPTION
**Pull request by ByteMeAsap**
This module allows to take online signatures from vendors to confirm purchase orders similar to how it is currently for sale orders.

### CODE REVIEW

**Comment by ByteMeAsap** [comment]
@rafaelbn , @yajo , @EmilioPascual , can you please review this

**Review by EmilioPascual — COMMENTED** [review]
Thank you @ByteMeAsap for your work.
I've test module in runboat but I can't get it to work.
I have made a purchase order with a supplier using a portal user.
image
But no purchase order is displayed on the portal.
image
Also, does it make sense for suppliers to access a portal to confirm their sales? Normally it would be the other way around, the buyer would have to access the supplier's platform.
Perhaps you could add context to the pr by commenting on the situation in which this flow could be used.

**Comment by ByteMeAsap** [comment]
> But no purchase order is displayed on the portal.

@EmilioPascual , only the RFQs in sent state would show up under Request for Quotation(as that's how its setted up in Odoo base). Once the RFQ is sent (by email) to the vendor, it would show up on the portal.

> Also, does it make sense for suppliers to access a portal to confirm their sales?

Yes, it does make sense in certain scenarios , for example we can define our rates for products to be purchased and ask suppliers for confirming the same or add their feedback through the communication section.

**Review by EmilioPascual — APPROVED** [review]

**Review by Jan-Onestein — APPROVED** [review]
thanks for the great work @ByteMeAsap

**Comment by ByteMeAsap** [comment]
@rafaelbn , @yajo , @EmilioPascual , can we have this merged?

File: purchase_sign/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+        try:
+            order_sudo.write(
+                {
+                    "signed_by": name,
+                    "signed_on": fields.Datetime.now(),
+                    "signature": signature,
+                }
+            )
+            request.env.cr.commit()
```
Comment (yajo) [inline purchase_sign/controllers/main.py]: issue: this is a [discouraged behavior](https://www.odoo.com/documentation/17.0/contributing/development/coding_guidelines.html#never-commit-the-transaction), please remove the commit.

**Review by yajo — CHANGES_REQUESTED** [review]

File: purchase_sign/controllers/main.py (hunk 1 above)
Comment (ByteMeAsap) [reply purchase_sign/controllers/main.py]: @yajo , I have removed the commit

**Review by yajo — APPROVED** [review]
/ocabot merge nobump

**Comment by Jan-Onestein** [comment]
@rafaelbn , @yajo , @EmilioPascual , can we please have this merged?

**Pull request merged**